lib/rubygems/psych_tree.rb

```diff
@@ -14,6 +14,10 @@ def visit_String(str)
         @emitter.scalar str, nil, nil, false, true, quote
       end
 
+      def visit_Hash(o)
+        super(o.dup.delete_if {|_, v| v.nil? })
```

Comment:
```suggestion
        super(o.compact)
```

since `Hash#compact` is available on Ruby 3+ ?